plugins/salesforce/src/org/pentaho/di/trans/steps/salesforce/SalesforceConnection.java

```diff
@@ -273,6 +273,9 @@ public void setPassword( String value ) {
   public void connect() throws KettleException {
     ConnectorConfig config = new ConnectorConfig();
     config.setAuthEndpoint( getURL() );
+    config.setServiceEndpoint( getURL() );
+    config.setUsername( getUsername() );
+    config.setPassword( Encr.decryptPasswordOptionallyEncrypted( getPassword() ) );
```

Comment:
Quick note about this - it is best practices to leave the password encrypted until the last possible moment where it needs to be decrypted. Your code here unfortunately decrypts the password into the data structure which would allow anyone to inspect/dump memory and see the password. The better answer is what used to be done before where the password was decrypted at the exact moment needed.